internal/test/wsu/wsu_test.go

```diff
@@ -196,27 +368,37 @@ func testFilesCopied(t *testing.T, vm e2ef.WindowsVM) {
 		"kube.tar.gz downloaded does not match expected checksum")
 }
 
-// testNodeReady tests that the bootstrapped node was added to the cluster and is in the ready state
-func testNodeReady(t *testing.T, vmCredentials *types.Credentials) {
-	var createdNode *v1.Node
+// getNode returns a pointer to the node object associated with the external IP provided
+func getNode(externalIP string) (*v1.Node, error) {
```

Comment:
It seems that this should become a common function. I will need something similar in my current work. So please move this into the framework.